### PR TITLE
access log: add GRPC_STATUS_NUMBER format operator

### PR DESCRIPTION
**Pull request by jmuia**
Similar to https://github.com/envoyproxy/envoy/issues/9352 but number format. Useful because `grpc-status` may appear in either the headers or in the trailers and there's no alternative to coalesce the value into a single access log field.

Appreciate feedback regarding C++ and testing. The change is simple enough, but I'm not particularly familiar with C++.

Risk Level: Low
Testing: Unit tests
Docs Changes: Required
Release Notes: Required

### CODE REVIEW

**Comment by wrowe** [comment]
/wait on fixes to resolve test/common/access_log/access_log_impl_test breakage

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ Minor Behavior Changes
 ----------------------
 *Changes that may cause incompatibilities for some users, but should not for most*
 
+* access_log: add GRPC_STATUS_NUMBER format command operator.
```
Comment (zuercher) [inline docs/root/version_history/current.rst:15]: I think this belongs farther down in the "New Features" section. You can see there are already some other new log operators there.

File: source/common/formatter/substitution_formatter.h [hunk 2]
```diff
@@ -411,6 +411,9 @@ class GrpcStatusFormatter : public FormatterProvider, HeaderFormatter {
   ProtobufWkt::Value formatValue(const Http::RequestHeaderMap&, const Http::ResponseHeaderMap&,
                                  const Http::ResponseTrailerMap&, const StreamInfo::StreamInfo&,
                                  absl::string_view) const override;
+
+private:
+  bool format_as_number_;
```
Comment (zuercher) [inline source/common/formatter/substitution_formatter.h:416]: Let's mark this `const`.

**Review by zuercher — COMMENTED** [review]
Thanks! That looks good to me, just two small requests.

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (jmuia) [reply docs/root/version_history/current.rst]: whoops – fixed!

File: source/common/formatter/substitution_formatter.h (hunk 2 above)
Comment (jmuia) [reply source/common/formatter/substitution_formatter.h]: fixed, thanks!

**Review by zuercher — APPROVED** [review]
Thanks!

**Pull request merged**